pkg/devfile/adapters/common/command.go

```diff
@@ -30,54 +28,54 @@ func getCommand(data data.DevfileData, commandName string, groupType common.Devf
 		// if not found fallback to error.
 		if commandName != "" {
 
-			if command.Exec.Id == commandName {
+			// Update Group only custom commands (specified by odo flags)
+			command = updateGroupforCommand(groupType, command)
 
-				if command.Exec.Group.Kind == "" {
-					// Devfile V1 for commands passed from flags
-					// Group type is not updated during conversion
-					command.Exec.Group.Kind = groupType
-				}
+			if command.Exec.Id == commandName {
 
 				// we have found the command with name, its groupType Should match to the flag
 				// e.g --build-command "mybuild"
 				// exec:
 				//   id: mybuild
-				// group:
-				//   kind: build
+				//   group:
+				//     kind: build
 				if command.Exec.Group.Kind != groupType {
-					return supportedCommand, fmt.Errorf("mismatched group kind, command %s is of group kind %v groupType in devfile", commandName, groupType)
-
+					return supportedCommand, fmt.Errorf("command group mismatched, command %s is of group %v in devfile.yaml", commandName, command.Exec.Group.Kind)
 				}
 				supportedCommand = command
 				return supportedCommand, nil
 			}
 			continue
 		}
 
-		// if not command specified via flag, default command has the highest priority
+		// if no command specified via flag, default command has the highest priority
 		// We need to scan all the commands to find default command
-		if command.Exec.Group.Kind == groupType && command.Exec.Group.IsDefault {
-			supportedCommand = command
-			return supportedCommand, nil
+		// exec.Group is a pointer, to avoid null pointer
+		if command.Exec.Group != nil {
+			if command.Exec.Group.Kind == groupType && command.Exec.Group.IsDefault {
```

Comment:
why not join the two lines using `&&`?

Reply:
if group is `nil` and we try to retrieve `kind`, it would give `nil pointer dereference `

Reply:
I guess it will short circuit if group is `nil` and it won't try to retrieve `Kind`. https://stackoverflow.com/questions/45698298/does-go-excecute-all-conditions-in-if-statement

Reply:
ya you are right, good to know, but do you think it is good to change? as someday may forget to preserve the order.

Reply:
You are right that somebody might forget to preserve the order, but we can have proper unit tests to prevent that scenario.